pkg/cp/cmp/cmp.go

```diff
@@ -56,27 +59,43 @@ func stat(o cp.Options, path string) (os.FileInfo, error) {
 	return os.Stat(path)
 }
 
-// IsEqualTree compare the content in the file trees in src and dst paths
-func IsEqualTree(o cp.Options, src, dst string) error {
-	srcInfo, err := stat(o, src)
+// For testing purpose
+var readLink = os.Readlink
```

Comment:
not sure why you are doing this, but the more accepted way to do it is to actually create a tree in a tmpdir (you are doing that below) and create symlinks in it.